src/storage/src/hummock/store/version.rs

```diff
@@ -819,4 +803,169 @@ impl HummockVersionReader {
         )
         .into_stream())
     }
+
+    // Note: this method will not check the kv tomestones and delete range tomestones
+    pub async fn surely_not_have(
+        &self,
+        table_key_range: TableKeyRange,
+        read_options: ReadOptions,
+        read_version_tuple: (Vec<ImmutableMemtable>, Vec<SstableInfo>, CommittedVersion),
+    ) -> StorageResult<bool> {
+        let table_id = read_options.table_id;
+        let table_id_string = table_id.to_string();
+        let table_id_label = table_id_string.as_str();
+        let mut table_counts = 0;
+        let mut local_stats = StoreLocalStatistic::default();
+        let (imms, uncommitted_ssts, committed_version) = read_version_tuple;
+
+        // 1. check staging data
+        for imm in &imms {
+            if imm.range_exists(&table_key_range) {
```

Comment:
only for this case, local_stats had not collected any metrics, we no need to report it ? Even if the report function has defensive logic

Reply:
good catch. removed.

Reply:
For the same reason, we also do not need to `report`